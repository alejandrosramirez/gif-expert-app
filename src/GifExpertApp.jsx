import { useState } from "react";

export const GifExpertApp = () => {
	const [categories, setCategories] = useState(["One Punch"]);

	return (
		<>
			<h1>GifExpertApp</h1>

			<ol>
				{
					categories.map((category, idx) => {
						return <li key={idx}>{category}</li>
					})
				}
			</ol>
		</>
	);
};
