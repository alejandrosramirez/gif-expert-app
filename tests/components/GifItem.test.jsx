import { render, screen } from "@testing-library/react";
import { GifItem } from "../../src/components";

describe("Pruebas en <GifItem />", () => {
	const title = "Pokémon";
	const url = "https://via.placeholder.com/150";

	test("Debe de hacer match con el snapshot", () => {
		const { container } = render(<GifItem title={title} url={url} />);
		expect(container).toMatchSnapshot();
	});

	test("Debe de mostrar el gif con el URL y el ALT indicado", () => {
		render(<GifItem title={title} url={url} />);
		const {src, alt} = screen.getByRole("img");
		expect(src).toBe(url);
		expect(alt).toBe(title);
	});

	test("Debe de mostrar el título en el componente", () => {
		render(<GifItem title={title} url={url} />);
		expect(screen.getByText(title)).toBeTruthy();
	});
});
