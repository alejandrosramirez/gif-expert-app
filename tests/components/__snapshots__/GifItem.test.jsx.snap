// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifItem /> Debe de hacer match con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Pokémon"
      src="https://via.placeholder.com/150"
    />
    <p>
      Pokémon
    </p>
  </div>
</div>
`;
